Correct orders of property chains

nmtigor opened the pull request:
There is no property `id` for `Operators`, and number `id` has no operator properties `dot`, `dotHash`,  `lt`, `or`. (see `Operators.lt.id` in `getOperatorOrValue()` below)

calixteman committed:
Use AccentColor as background for selected text in the text layer (bug 1790309)

and use the default color for the viewer in general.

nmtigor committed:
Fix orders of property chains of Operators in AstBinaryOperator's methods.

calixteman committed:
[Editor] Make sure to have the annotation editor layer on top of the annotation one (bug 1791515)

Some z-index have been added in the annotation layer because the elements inside are re-ordered
in order to improve accessibility.
Hence we must add a "high" z-index on the annotation editor layer in order to avoid any bad
interaction between the different layers.

Snuffleupagus commented:
(Again, a slightly better commit message wouldn't hurt here.)

This seems like the sort of change that really needs a test-case, to both ensure that it works correctly now and in order to prevent any future regressions.

nmtigor committed:
Fix commFunc typo in the SandboxSupportBase.destroy method

nmtigor committed:
Fix #selectEditors typo in the AnnotationEditorUIManager.unselectAll method

calixteman committed:
[XFA] Fix an hidden issue in the FormCalc lexer

Since there are no script engine with XFA, the FormCalc parser is not used irl.
The bug @nmtigor noticed was hidden by another one (the wrong check on `match`).

Snuffleupagus committed:
Access the `setTimeout`-functionRefs correctly in `SandboxSupportBase.destroy`

*This effectively replaces PR 15465.*

As outlined in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map/forEach, the argument order when iterating through a `Map` is actually `value, key`.
Ignoring the incorrect Array used in the old code, I cannot imagine that this would've worked anyway since we didn't use the actual `setTimeout`-functionRefs to clear the timeouts; please refer to the `setTimeout`/`setInterval` methods in the `SandboxSupportBase.createSandboxExternals` method.

calixteman committed:
Initialize values in the path bounding box before flushing the operator list (bug 1791583)

OperatorList.addOp can trigger a flush if it's required, hence the values passed to it must
be correctly initialized in order to avoid some wrong values in the renderer.
Because of that a clip path was considered as empty, nothing was clipped, hence the wrong
rendering in bug 1791583.

Snuffleupagus commented:
This PR contains lots of unrelated commits (it looks like a rebase gone wrong), so those must be removed.
Also, as mentioned in https://github.com/mozilla/pdf.js/pull/15467#issuecomment-1252089687, this definitely seems like the sort of change where tests would be good to have.

calixteman commented:
About testing, you just have to add a test in [xfa_form_spec.js](https://github.com/mozilla/pdf.js/blob/master/test/unit/xfa_formcalc_spec.js).
You can run unit tests in using the command `gulp unittestcli` or `gulp unittest`. You can prepend a "f" on `describe` or `it` in order to focus on a test.
